app/controllers/spree/amazon_controller.rb

```diff
@@ -29,7 +29,8 @@ def payment
     payment.payment_method = gateway
     payment.source ||= Spree::AmazonTransaction.create(
       order_reference: params[:order_reference],
-      order_id: current_order.id
+      order_id: current_order.id,
+      retry: current_order.amazon_transactions.where(authorization_success: false).any?
```

Comment:
might want to create a scope for `where(authorization_success: false)` - can call it `unsuccessful`